crates/subspace-farmer/src/plotting.rs

```diff
@@ -302,15 +301,23 @@ async fn background_plotting<T: RpcClient + Clone + Send + 'static>(
                 info!("Plotting stopped!");
                 break;
             }
-            Some(head) = new_head.recv() => {
-                let block_number = u32::from_str_radix(&head.number[2..], 16).unwrap();
-                debug!("Last block number: {:#?}", block_number);
-
-                if let Some(block) = block_number.checked_sub(confirmation_depth_k) {
-                    // We send block that should be archived over channel that doesn't have a buffer, atomic
-                    // integer is used to make sure archiving process always read up to date value
-                    block_to_archive.store(block, Ordering::Relaxed);
-                    let _ = new_block_to_archive_sender.try_send(Arc::clone(&block_to_archive));
+            result = new_head.recv() => {
+                match result {
+                    Some(head) => {
+                        let block_number = u32::from_str_radix(&head.number[2..], 16).unwrap();
+                        debug!("Last block number: {:#?}", block_number);
+
+                        if let Some(block) = block_number.checked_sub(confirmation_depth_k) {
+                            // We send block that should be archived over channel that doesn't have a buffer, atomic
+                            // integer is used to make sure archiving process always read up to date value
+                            block_to_archive.store(block, Ordering::Relaxed);
+                            new_block_to_archive_sender.try_send(Arc::clone(&block_to_archive)).expect("Cannot send the message to plot!");
```

Comment:
This is incorrect, result was explicitly and intentionally ignored because channel can and will be full sometimes. Now this is causing crashes.